src/main/python/rlbot/gui/presets.py

```diff
@@ -0,0 +1,100 @@
+import os
+from PyQt5.QtCore import QTimer
+import configparser
+
+from rlbot.agents.base_agent import BaseAgent
+from rlbot.agents.base_agent import PYTHON_FILE_KEY, BOT_CONFIG_MODULE_HEADER
+from rlbot.utils.class_importer import import_agent, get_python_root
+
+
+class Preset:
+    def __init__(self, config, file_path, name):
+        self.config = config
+        self.config_path = file_path
+        self.name = name
+        self.save_loadout_timer = None
+        self.auto_save = True
+        if file_path is not None:
+            self.load(file_path)
+
+    def load(self, file_path=None):
+        if file_path is None:
+            file_path = self.config_path
+        if not os.path.isfile(file_path):
+            self.config_path = ""
+            return
+        self.config_path = file_path
+        raw_parser = configparser.RawConfigParser()
+        raw_parser.read(file_path)
+        for section in self.config.headers.keys():
+            if not raw_parser.has_section(section):
+                raise configparser.NoSectionError(section)
+        self.config.parse_file(raw_parser)
+        return self.config
+
+    def save_config(self, file_path=None, time_out=0):
+        if file_path is not None:
+            self.config_path = file_path
+
+        def save():
+            if self.config_path is None or self.config_path == "":
+                return
+            with open(self.config_path, "w") as f:
+                f.write(str(self.config))
+
+        if self.save_loadout_timer is None:
+            self.save_loadout_timer = QTimer()
+            self.save_loadout_timer.setSingleShot(True)
+            self.save_loadout_timer.timeout.connect(save)
+        self.save_loadout_timer.start(time_out)  # Time-out for timer over here
+
+    def set_auto_save(self, auto_save):
+        self.auto_save = auto_save
+
+    def get_name(self):
+        return self.name
+
+
+class LoadoutPreset(Preset):
```

Comment:
can we put this in a separate file too (I know it will be tiny but it will be cleaner)